microcloud/cmd/microcloud/main_init.go

```diff
@@ -414,6 +419,68 @@ func AddPeers(sh *service.Handler, systems map[string]InitSystem) error {
 	return nil
 }
 
+func validateSystems(s *service.Handler, systems map[string]InitSystem) (err error) {
+	curSystem, bootstrap := systems[s.Name]
+	if !bootstrap {
+		return nil
+	}
+
+	// Assume that the UPLINK network on each system is the same, so grab just
+	// the gateways from the current node's UPLINK to verify against the other
+	// systems' management net addrs
+	hasIP4Gateway, hasIP6Gateway := false, false
+	var ip4Gateway, ip6Gateway string
+	var ip4GatewayNet, ip6GatewayNet *net.IPNet
+
+	for _, network := range curSystem.Networks {
+		if network.Type == "physical" && network.Name == "UPLINK" {
+			ip4Gateway, hasIP4Gateway = network.Config["ipv4.gateway"]
+			ip6Gateway, hasIP6Gateway = network.Config["ipv6.gateway"]
+			ovnRanges, hasOVNRanges := network.Config["ipv4.ovn.ranges"]
+
+			if hasIP4Gateway {
+				_, ip4GatewayNet, err = net.ParseCIDR(ip4Gateway)
```

Comment:
The gateway fields are expecting an IP address and subnet in CIDR format.
This check as it stands would allow an invalid entry where the address is the network address of the subnet, e.g.

`192.168.1.0/24` should be invalid, but `192.168.1.1/24` is valid.

See `IsNetworkAddressCIDRV4()` in LXD.